mantidproject/mantid
# Algorithm to save diffraction fitting results

**Description of work.**

`SaveDiffFittingAscii` algorithm will convert a `TableWorkspace` containing diffraction fitting results into an ASCII file format. All of the columns must be typed as double in order to successfully save it as a ASCII file. A `GroupWorkspace` of `TableWorkspace` can also be saved when a list of run and bank numbers is provided manually. 

**To test:**

_Test 1_
Go to: `Interfaces/Diffraction/Engineering Diffraction/Fitting Tab`
`Focus Run #`: `ENGINX_241391_focused_texture_bank_3.nxs` 
Focused file can be found here: `\\olympic\Babylon5\Public\Shahroz\FittingTest\ENGINX_241391_focused_texture_bank_3.nxs`
Enter Peaks Manually: `1.1437,1.2526,1.2849,1.9807,2.8022,3.2357,1.4011,1.6917,1.6171`
Click `Fit`

Upon completion:

Run Algorithm: `SaveDiffFittingAscii` with `engggui_fitting_fitpeaks_param` table workspace
Run Number: `241391`
Bank: `3`
- [x] Check that the file has been saved in the correct format.

Read documentation for further check.

_Test 2_
duplicate `engggui_fitting_fitpeaks_param` table workspace and group the two table workspaces
Run algorithm with `GroupWorkspace`
Run Number: `241391, 241392`
Bank: `1, 3`
OutFormat: `WriteGroupWorkspace`
- [x] Check that both table workspaces have been written to the file in the correct format

_Review_
Documentation page
Doc test



[Release Notes](https://github.com/mantidproject/mantid/blob/b7be73a2d42de5e9f0a37741f7a9073d77087033/docs/source/release/v3.8.0/diffraction.rst)
Fixes #17048 



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
